autoload/leaderf/python/leaderf/devicons.py

```diff
@@ -0,0 +1,200 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+import re
+import vim
+from functools import wraps
+from .utils import *
+
+fileNodesDefaultSymbol = lfEval("get(g:, 'WebDevIconsUnicodeDecorateFileNodesDefaultSymbol', '')")
+
+fileNodesExactSymbols = lfEval("get(g:, 'WebDevIconsUnicodeDecorateFileNodesExactSymbols', {})")
+fileNodesExtensionSymbols = lfEval("get(g:, 'WebDevIconsUnicodeDecorateFileNodesExtensionSymbols', {})")
+
+_ambiwidth = lfEval('&ambiwidth')
+
+_iconBytesLen = 0
+
+_default_palette = {
+    "gui": "NONE",
+    "guifg": "NONE",
+    "guibg": "NONE",
+    "cterm": "NONE",
+    "ctermfg": "NONE",
+    "ctermbg": "NONE",
+}
+
+RE_CANNOT_USE_FOR_HIGHLIGHT = re.compile(r'[^a-zA-Z0-9_]+')
+
+def _icons_setup():
+    symbols = set()
+    symbols = symbols.union(set(fileNodesExactSymbols.values()))
+    symbols = symbols.union(set(fileNodesExtensionSymbols.values()))
+    symbols.add(fileNodesDefaultSymbol)
+
+    names = set()
+    names = names.union(set(fileNodesExactSymbols.keys()))
+    names = names.union(set(fileNodesExtensionSymbols.keys()))
+    names.add('default')
+
+    return {'names': names, 'symbols': symbols}
+
+
+_icons = _icons_setup()
+
+
+def removeDevIcons(func):
+    @wraps(func)
+    def deco(*args, **kwargs):
+        if vim.vars.get('Lf_ShowDevIcons', True) and 'loaded_webdevicons' in vim.vars:
+            is_list = isinstance(args[1], list)
+
+            lines = args[1] if is_list else [args[1]]
+
+            res_lines = []
+            for line in lines:
+                b_line = lfByteArray(line)
+                b_line = b_line[webDevIconsBytesLen():]
+                line = lfBytes2Str(b_line, encoding="utf-8")
+                res_lines.append(line)
+
+            _args = list(args)
+            _args[1] = res_lines if is_list else res_lines[0]
+
+            args = tuple(_args)
+        return func(*args, **kwargs)
+    return deco
+
+def _getExt(file):
+    idx = file.rfind('.')
+    return '' if idx == -1 else file[idx+1:]
+
+def setAmbiwidth(val):
+    global _ambiwidth
+    _ambiwidth = val
+
+# from vim-devicons
+# To use asynchronously
+def webDevIconsGetFileTypeSymbol(file):
+    if 'loaded_webdevicons' not in vim.vars:
```

Comment:
this line causes error when using neovim, I want to remove the  vim-devicons dependence. Or remove this line and I'll merge the PR, I'll remove the vim-devicons dependence later.

Reply:
I want to keep the vim-devicons dependencies because every time a new icon is added to vim-devicons, I have to add it to LeaderF as well.
Isn't that right?

Therefore, I would like to amend it as follows.

```diff
diff --git a/autoload/leaderf/python/leaderf/devicons.py b/autoload/leaderf/python/leaderf/devicons.py
index 49c1720..40068fb 100644
--- a/autoload/leaderf/python/leaderf/devicons.py
+++ b/autoload/leaderf/python/leaderf/devicons.py
@@ -46,7 +46,7 @@ _icons = _icons_setup()
 def removeDevIcons(func):
     @wraps(func)
     def deco(*args, **kwargs):
-        if vim.vars.get('Lf_ShowDevIcons', True) and 'loaded_webdevicons' in vim.vars:
+        if lfEval("get(g:, 'Lf_ShowDevIcons', '1')") == "1" and lfEval("has_key(g:, 'loaded_webdevicons')") == "1":
             is_list = isinstance(args[1], list)

             lines = args[1] if is_list else [args[1]]
@@ -76,7 +76,7 @@ def setAmbiwidth(val):
 # from vim-devicons
 # To use asynchronously
 def webDevIconsGetFileTypeSymbol(file):
-    if 'loaded_webdevicons' not in vim.vars:
+    if lfEval("has_key(g:, 'loaded_webdevicons')") == "0":
         return ''
     fileNode = getBasename(file).lower()
     fileNodeExt = _getExt(fileNode)
```

Reply:
I still want to remove the dependence in the future. 
The patch does not resolve the error in neovim. The solution can be :
`_loaded_webdevicons = 'loaded_webdevicons' in vim.vars` 
evaluate it in advance.
```python
def webDevIconsGetFileTypeSymbol(file):
    if not _loaded_webdevicons:
        return ''
```

Reply:
I understand. There is no problem in the direction of removing dependency in the future.
Thank you for your advice.
I fixed it.